src/parser/jobs/brd/modules/RagingStrikes.tsx

```diff
@@ -218,4 +309,27 @@ export class RagingStrikes extends BuffWindow {
 		const firstGcd = gcdTimestamps[0]
 		return this.barrageRemoves.some(timestamp => firstGcd <= timestamp && timestamp <= (window.end ?? window.start))
 	}
+
+	private wasActionUsedInTime = (action: ActionKey, bestBeforeGCD: number, window: HistoryEntry<EvaluatedAction[]>): boolean => {
+		const buffAction = _.first(window.data.filter(it => it.action.id === this.data.actions[action].id))
+
+		if (buffAction) {
+			const gcdThreshold = _.last(
+				window.data
+					.filter(it => it.action.onGcd)
+					.slice(0, bestBeforeGCD)
+			)
+
+			if (gcdThreshold) {
+				// Used in time if it was used before specified GCD
+				return buffAction.timestamp < gcdThreshold.timestamp
+			}
+
+			// There's no limiting GCD in the window, so can only assume it was used in time
+			return true
+		}
+
+		// Buff was not used, thus it was not in time
+		return false
```

Comment:
This is very nitpicky but to avoid the nesting you can invert the `buffAction` check to bail early instead of returning false at the very end. Something like this: 

```ts
if (buffAction == null) { return false }

const gcdThreshold = ...
```